Added properties for CubeMap.

NathanWarden opened the pull request:
Changed the methods that get or set a single value into properties for CubeMap.

NathanWarden committed:
Added properties for CubeMap.

Noshyaar commented on scene/resources/texture.cpp:
```diff
@@ -1324,17 +1324,19 @@ void CubeMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_lossy_storage_quality"), &CubeMap::get_lossy_storage_quality);
 
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "flags"), "set_flags", "get_flags");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "storage_mode", PROPERTY_HINT_ENUM, "STORAGE_RAW, STORAGE_COMPRESS_LOSSY, STORAGE_COMPRESS_LOSSLESS"), "set_storage", "get_storage");
```
maybe `"Raw,Lossy Compressed,Lossless Compressed"` ?

NathanWarden replied:
That sounds good, thanks.

cbscribe commented:
Sorry about that - clicked on the wrong PR!

Noshyaar commented on scene/resources/texture.cpp:
```diff
@@ -1324,17 +1324,19 @@ void CubeMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_lossy_storage_quality", "quality"), &CubeMap::set_lossy_storage_quality);
 	ClassDB::bind_method(D_METHOD("get_lossy_storage_quality"), &CubeMap::get_lossy_storage_quality);
 
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "flags"), "set_flags", "get_flags");
```
this is also enum

NathanWarden replied:
Will fix, thanks :)

Noshyaar replied:
apologize, it should use `PROPERTY_HINT_FLAGS`, not enum

Noshyaar commented on scene/resources/texture.cpp:
```diff
@@ -1324,17 +1324,19 @@ void CubeMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_flags", "flags"), &CubeMap::set_flags);
 	ClassDB::bind_method(D_METHOD("get_flags"), &CubeMap::get_flags);
-
 	ClassDB::bind_method(D_METHOD("set_side", "side", "image"), &CubeMap::set_side);
```
this could be properties too if im not mistaken. see `ADD_PROPERTYI` in control.cpp for example

NathanWarden replied:
Looking at it it could, but looking at Control.cpp you have to then make a separate method for each item. For GUI anchors it probably makes more sense than a cube map. In this particular case I would recommend against it, or have both a method and properties.

For instance, instead of:
``` cpp
for (int i = 0; i < 6; i++)
{
    cubemap_sides[i] = get_side(i);
    // do work on cubemap_sides[i]
    set_side(i, cubemap_sides[i]);
}
```

You have to:
``` cpp
cubemap_sides[0] = top_side;
cubemap_sides[1] = bottom_side;
cubemap_sides[2] = left_side;
cubemap_sides[3] = right_side;
cubemap_sides[4] = front_side;
cubemap_sides[5] = back_side;

for ( int i = 0; i < 6; i++ )
{
     // do work on cubemap_sides[i]
}

top_side = cubemap_sides[0];
bottom_side = cubemap_sides[1];
left_side = cubemap_sides[2];
right_side = cubemap_sides[3];
front_side = cubemap_sides[4];
back_side = cubemap_sides[5];
```

It would also make writing the code more difficult when you want to do thing like update one side per frame for performance reasons.

What do you think?

NathanWarden replied:
Thinking about it again, I think it would be good to have both. I'll be out for a little bit and will implement the PROPERTYI.

NathanWarden commented:
@cbscribe Funny thing is that the reason this PR exists is because I was going to document it, then realized the get/set methods would be changed to properties in the future, then just decided to do the properties myself. So docs will be coming soon for it too :D

NathanWarden commented:
Okay, I added PROPERTYI bindings for each image side now also, but also added a way to get/set each side by its' index since that works better for more algorithmic code such as only updating one side per frame or getting/setting images in a for loop.

akien-mga commented:
Too late, I have merged it already.

NathanWarden commented:
Oops, haha. It wasn't anything too important :)
